### PR TITLE
BUG: TimedeltaArray+Period, PeriodArray-TimedeltaArray

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

motivated by the FIXME comment in test_timedelta64.py

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Can you explain what this PR is changing?

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note.

**Comment by jreback** [comment]
can you also add an example in the docs (maybe in the timedelta section)

**Comment by jbrockmendel** [comment]
> Can you explain what this PR is changing?

PeriodIndex - TimedeltaIndex always raises in master, but for some `freq` combinations should not.  This fixes that.

File: pandas/tests/arithmetic/test_period.py [hunk 1]
```diff
@@ -728,13 +728,13 @@ def test_pi_add_sub_td64_array_non_tick_raises(self):
         tdi = pd.TimedeltaIndex(["-1 Day", "-1 Day", "-1 Day"])
         tdarr = tdi.values
 
-        msg = r"Input has different freq=None from PeriodArray\(freq=Q-DEC\)"
-        with pytest.raises(IncompatibleFrequency, match=msg):
+        msg = r"Cannot add or subtract timedelta64\[ns\] dtype from period\[Q-DEC\]"
+        with pytest.raises(TypeError, match=msg):
```
Comment (jreback) [inline pandas/tests/arithmetic/test_period.py:732]: doesn't IncompatibleFrequency inherit from ValueError? so this is technically an api change (ok though)

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. can you add a whatsnew note.

File: pandas/tests/arithmetic/test_period.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/arithmetic/test_period.py:734]: @jbrockmendel ?

Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_period.py:734]: yes, this is a change.  in the whatsnew i called it a bugfix

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**